### ISSUE TITLE
Ability to identify element children either of complex or sequence or etc

### ISSUE THREAD

**Issue by TheSachin** [goetas-webservices/xsd-reader#44]
Hi,
Is there a way to identify if the children of the element are from sequence or choice or any other tag.
Right now, I am able to get the children but not the type. Is this in the scope of the project?
Here is an example.
`  <xsd:complexType name="Parent">

        <xsd:sequence>

            <xsd:element name="child1" type="wsg:child1" minOccurs="0" />

            <xsd:element name="child2" type="wsg:child2" />

            <xsd:element name="child3" type="wsg:child3" minOccurs="0" />

        </xsd:sequence>

    </xsd:complexType>`

Basically, I would prefer If i could get that it's type sequence or choice or something similar.

**Comment by leuchtdiode** [comment]
I'm curious too. Is this possible?

**Comment by goetas** [comment]
In theory is possible, but currently not implemented.

**Comment by leuchtdiode** [comment]
Do you have plans on implementing it (any time soon)? :)

**Comment by goetas** [comment]
Not really.. I did not have a use case for it.

On top of that, the concept of choice/sequence is not really something that fits well in the php object model (that is one of the reasons for not implementing it). 

Do you have an idea on how this could be implemented?

**Comment by leuchtdiode** [comment]
I have to tackle this problem in one of my projects. So if I come up with a good solution I'll let you know.

**Comment by MrMadClown** [comment]
Sorry for reviving this, 

I started using this lib today and stumbled across this problem as well.

Wouldn't `Sequence`, `Choice` and `All` just be another ElementContainer?

vendor/goetas-webservices/xsd-reader/src/SchemaReader.php:592
``` switch ($childNode->localName) {
            case 'sequence':
            case 'choice':
            case 'all':
                if ($type instanceof ElementContainer) {
                    $this->loadSequence(
                        $type,
                        $childNode
                    );
                }
                break;
```




``` switch ($childNode->localName) {
            case 'sequence':
                if ($type instanceof ElementContainer) {
                  $sequence = new Sequence();
                  $type->addElement($sequence);
                    $this->loadSequence(
                        $sequence,
                        $childNode
                    );
                }
                break;
            case 'choice':
```

@goetas Something like that?

I have a case where a ComplexType contains multiple nested `sequence` and some `choice`/`all`- nodes as well. Which this lib currently just flattens..

**Comment by Guite** [comment]
Proper choice support has been added in 0.4.0.
